packages/core/src/render3/instructions/shared.ts

```diff
@@ -863,28 +863,39 @@ export function createTNode(
 
 
 function generatePropertyAliases(
-    inputAliasMap: {[publicName: string]: string}, directiveDefIdx: number,
-    propStore: PropertyAliases|null): PropertyAliases|null {
+    inputAliasMap: {[publicName: string]: string}, directiveIndex: number,
+    propStore: PropertyAliases|null,
+    aliasMap: {[internalName: string]: string}|null): PropertyAliases|null {
   for (let publicName in inputAliasMap) {
     if (inputAliasMap.hasOwnProperty(publicName)) {
       propStore = propStore === null ? {} : propStore;
       const internalName = inputAliasMap[publicName];
 
-      if (propStore.hasOwnProperty(publicName)) {
-        propStore[publicName].push(directiveDefIdx, internalName);
-      } else {
-        (propStore[publicName] = [directiveDefIdx, internalName]);
+      if (aliasMap === null) {
+        addToPropStore(propStore, directiveIndex, publicName, internalName);
+      } else if (aliasMap.hasOwnProperty(publicName)) {
+        addToPropStore(propStore, directiveIndex, aliasMap[publicName], internalName);
       }
```

Comment:
Proposal to improve readability (and a request to add some docs):
```suggestion
      if (aliasMap !== null && aliasMap.hasOwnProperty(publicName)) {
        // ... docs on why we do it ...
        publicName = aliasMap[publicName];
      }
      addToPropStore(propStore, directiveIndex, publicName, internalName);
```

Reply:
This doesn't cover the same cases though. The host directive alias map has two purposes: it acts as an allowlist for the inputs/outputs and it remaps the bindings to a different name. In your example the allowlist isn't covered. I've added a comment explaining why it's written this way.